spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -109,6 +112,64 @@ object Engine {
   private var physicalCoreNumber = -1
   private var nodeNum: Int = -1
 
+  @volatile
+  private var gatewayServer: py4j.GatewayServer = null
+
+
+  private[bigdl] def createJavaGateway(driverPort: Int): Unit = {
+    if (SparkUtils.isDriver) {
+      val file = new java.io.File(SparkFiles.getRootDirectory(), "gateway_port")
```

Comment:
Should we make sure it's thread safe when check and write the file?

Reply:
Done